app/src/main/java/io/gnosis/safe/ui/safe/settings/SafeSettingsFragment.kt

```diff
@@ -0,0 +1,66 @@
+package io.gnosis.safe.ui.safe.settings
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.lifecycle.Observer
+import androidx.navigation.fragment.findNavController
+import io.gnosis.data.models.Safe
+import io.gnosis.safe.databinding.FragmentSafeSettingsBinding
+import io.gnosis.safe.di.components.ViewComponent
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import io.gnosis.safe.ui.safe.SafeOverviewBaseFragment
+import pm.gnosis.svalinn.common.utils.snackbar
+import javax.inject.Inject
+
+class SafeSettingsFragment : SafeOverviewBaseFragment<FragmentSafeSettingsBinding>() {
+
+    @Inject
+    lateinit var viewModel: SafeSettingsViewModel
+
+    override fun inject(component: ViewComponent) {
+        component.inject(this)
+    }
+
+    override fun inflateBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentSafeSettingsBinding =
+        FragmentSafeSettingsBinding.inflate(inflater, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.button.setOnClickListener {
+            viewModel.removeSafe()
```

Comment:
Just a thought, we should work on debounced clicks with coroutines. This could lead to duplicates when we get deeper with DAOs.